be1-go/internal/popserver/handler/root.go

```diff
@@ -0,0 +1,230 @@
+package handler
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"popstellar/crypto"
+	"popstellar/internal/popserver/config"
+	"popstellar/internal/popserver/database"
+	"popstellar/internal/popserver/database/sqlite"
+	"popstellar/internal/popserver/state"
+	"popstellar/message/answer"
+	"popstellar/message/messagedata"
+	"popstellar/message/query/method/message"
+)
+
+const (
+	Root       = "/root"
+	RootPrefix = "/root/"
+	Social     = "/social"
+	Chirps     = "/chirps"
+	Reactions  = "/reactions"
+	Consensus  = "/consensus"
+	Coin       = "/coin"
+	Auth       = "/authentication"
+)
+
+func handleChannelRoot(msg message.Message) *answer.Error {
+	object, action, errAnswer := verifyDataAndGetObjectAction(msg)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleChannelRoot")
+	}
+
+	switch object + "#" + action {
+	case messagedata.LAOObject + "#" + messagedata.LAOActionCreate:
+		errAnswer = handleLaoCreate(msg)
+	default:
+		errAnswer = answer.NewInvalidMessageFieldError("failed to handle %s#%s, invalid object#action", object, action)
+	}
+
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleChannelRoot")
+	}
+
+	return nil
+}
+
+func handleLaoCreate(msg message.Message) *answer.Error {
+	var laoCreate messagedata.LaoCreate
+	errAnswer := msg.UnmarshalMsgData(&laoCreate)
+	if errAnswer != nil {
+		return errAnswer.Wrap("handleLaoCreate")
+	}
+
+	laoPath := RootPrefix + laoCreate.ID
+	organizerPubBuf, errAnswer := verifyLaoCreation(msg, laoCreate, laoPath)
```

Comment:
why should the laoPath be passed when it can be created by the function (and even returned if valid) ?